File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.js
class Solution {
    /**
     * For each number, find the next number 
     * 
     * @param {number[]} nums
     * @return {number}
     */
    longestConsecutive(nums) {
        const numSet = new Set(nums)
        let longest = 0

        for (const num of numSet) {
            // Make sure current number is the smallest value of consecutive sequence
            // by making sure there's no smaller item
            if (numSet.has(num - 1)) continue
            let length = 1
            while (numSet.has(num + length)) length++
            longest = Math.max(length, longest)
        }

        return longest
    }
}

File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.go
func containsNearbyDuplicate(nums []int, k int) bool {
	dict := make(map[int]int)
	for i, num := range nums {
		if j, ok := dict[num]; ok && i-j <= k {
			return true
		}
		dict[num] = i
	}
	return false
}
